frontend/lib/resolution/Resolver.cpp

```diff
@@ -5052,6 +5126,7 @@ static QualifiedType resolveReduceScanOp(Resolver& resolver,
                                          const AstNode* reduceOrScan,
                                          const AstNode* op,
                                          const AstNode* iterand) {
+  iterand->traverse(resolver);
```

Comment:
I actually like it this way, particularly for loops, since for array declarations, we need to "separately" resolve the iterand, and doing so in `resolveIterDetails` would cause either duplicate resolution or tiptoeing.